tests/test_move.py

```diff
@@ -60,3 +63,91 @@ def test_move_dir(self):
         if self.preserve_time:
             dst_file2_info = dst_fs.getinfo("bar/baz.txt", namespaces)
             self.assertEqual(dst_file2_info.modified, src_file2_info.modified)
+
+
+class TestMove(unittest.TestCase):
+    def test_move_file(self):
+        with open_fs("temp://") as temp:
+            syspath = temp.getsyspath("/")
+            a = open_fs(syspath)
+            a.makedir("dir")
+            b = open_fs(join(syspath, "dir"))
+            b.writetext("file.txt", "Content")
+            fs.move.move_file(b, "file.txt", a, "here.txt")
+            self.assertEqual(a.readtext("here.txt"), "Content")
+            self.assertFalse(b.exists("file.txt"))
+
+    def test_move_file_different_mems(self):
```

Comment:
Feels like this ought to be renamed to `test_move_file`, modified to also check the `preserve_time`, and then moved to the `TestMovePreserveTime` class? (i.e. have it as the "base test")
And then rename `TestMove.test_move_file` to e.g. TestMove.test_move_file_optimised` or TestMove.test_move_file_tempfs` ? (since I guess you can't actually unit-test whether the optimised code-path is being followed).